.pre-commit-config.yaml

```diff
@@ -30,3 +26,10 @@ repos:
         files: 'Explorer/.*\.(js|jsx|ts|tsx)$'
         entry: bash -c 'cd Explorer && npm run precommit-lint "$@"'
         types: [file]
+      - id: golangci-lint
+        name: golangci-lint
+        language: system
+        types: [file, go]
+        # This checks all files instead of only the ones that were changed.
+        # If this check is taking too long we can refine this
+        entry: bash -c 'echo "$@"; RESULT=0;cd Proxy && for i in "$@"; do echo "$i" | cut -d'/' -f2- | xargs golangci-lint run --fix; ((RESULT|=$?)); done && exit $RESULT'
```

Comment:
This approach does not work with golangci-lint modules like typecheck. Is there a reason to not just run golangci-lint on the whole go package?

Reply:
Yeah, that's a better idea